source/include/azure_iot_version.h

```diff
@@ -0,0 +1,34 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file azure_iot_version.h
+ *
+ * @brief Provides version information.
+ *
+ * @note You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ */
+
+#ifndef AZURE_IOT_VERSION_H
+#define AZURE_IOT_VERSION_H
+
+/*/ The version in string format used for telemetry following the `semver.org` standard */
+/*/ (https://semver.org). */
+#define azureiotVERSION_STRING        "0.0.1-alpha"
```

Comment:
It'd be safer if we could derive `azureiotVERSION_STRING` from the `azureiotVERSION_MAJOR` and friends below.  This way there's only one source of truth instead of possibility for this to be split up.

Reply:
you mind giving the newest version another look? Just to make sure the macros are acceptable. This is how it renders:
![image](https://user-images.githubusercontent.com/36710865/114904343-9c884e80-9dcc-11eb-8313-b491bcce671d.png)

Reply:
Yup the latest around stringify stuff looks good.